### PR TITLE
Incorporate basic api

### PR DESCRIPTION
**Pull request by vincerubinetti**
- hookup basic api calls for looking up and querying genesets
- change base url from `/mygeneset.info-website` to `/`
- add copy button component
- add loading/error states to geneset edit page
- hookup geneset page to actual genesets from api
- improve table sorting method (fix bug where two columns with identical keys dont get sorted properly
- hookup actual api calls to "try it out" section
- refactor "try it out" component, add tooltips
- get rid of "GenesetEdit" HOC

### CODE REVIEW

**Comment by vincerubinetti** [comment]
- geneset page redirect broken
- `_exists_:kegg` query breaks results format

File: src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+export const request = async (url: string): Response => {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) throw new Error();
+    const json = await response.json();
+    if (typeof json === "object" && !Array.isArray(json)) return json;
+    else throw new Error();
+  } catch (error) {
+    throw new Error(`Couldn't complete request\n${url}`);
```
Comment (dongbohu) [inline src/api/index.ts:13]: Did you add "\n" in the middle on purpose? It seems a little weird.

**Review by dongbohu — APPROVED** [review]
Only left a trivial comment.

File: src/api/index.ts (hunk 1 above)
Comment (vincerubinetti) [reply src/api/index.ts:13]: Yes, this was to put the request url on a new line

**Pull request merged**